packages/trader/src/Modules/CFD/Containers/cfd-dashboard.tsx

```diff
@@ -117,26 +226,25 @@ const CFDDashboard = props => {
         setIsNotificationLoaded(true);
     };
 
-    const updateActiveIndex = index => {
-        if (!index) return;
-        const updated_state = {};
+    const updateActiveIndex = (index?: number) => {
+        const updated_state: { is_demo_tab?: boolean; active_index?: number } = {};
         // updateActiveIndex is called in componentDidUpdate causing tab_index to always revert back to 0
         if (index === 1) updated_state.is_demo_tab = true;
         else if (index === 0) updated_state.is_demo_tab = false;
 
-        if (active_index !== index) {
+        if (index !== undefined && active_index !== index) {
             updated_state.active_index = index;
         }
 
         if (!isEmptyObject(updated_state)) {
-            setActiveIndex(updated_state.active_index);
-            setIsDemoTab(updated_state.is_demo_tab);
+            setActiveIndex(updated_state.active_index!);
+            setIsDemoTab(updated_state.is_demo_tab!);
```

Comment:
I suggest removing '`!`'s at these 2 lines because `active_index` and `is_demo_tab` can actually be set to `undefined` here if none of the conditions are satisfied. no need to ignore it with '`!`'s 😊
Instead, I suggest changing the useState types for them at lines 149 & 150: 
```
const [active_index, setActiveIndex] = React.useState<number | undefined>(0);
const [is_demo_tab, setIsDemoTab] = React.useState<boolean | undefined>(true);
```

Reply:
our active_index should not be undefined to prevent further issues. Since we have the !isEmptyObject condition before setting the state we are sure that our updated_state is not empty before SetStateAction.

Reply:
@hirad-rewok !isEmptyObject(updated_state) is checking that updated_state has at least 1 property but it doesn't state which one. So for example, is_demo_tab can be present and active_index might not. Why did you remove my check for undefined index at line 121 that I added as part of fix https://github.com/binary-com/deriv-app/pull/4833 ?
Anyway, @maryiafrantsava-deriv will soon improve my check for undefined index that comes to this function in parameters in rare cases and causes severe consequences: https://github.com/binary-com/deriv-app/pull/4853/files
Once the check is added, the active_index will be never undefined.
I'm sorry to say that but there is an opinion that using `!` is a bad practice, that's why I suggest using an assertion if you don't like the idea to change the useState types for them:
```
setActiveIndex(updated_state.active_index as number);
setIsDemoTab(updated_state.is_demo_tab as boolean);
```
What do you think?)

Reply:
I suppose 19 code smells are too much for the refactoring tasks. if you need help, let me know

Reply:
The last pull that I had from the master removed some of the lines that you are mentioning. I will make another branch with the included changes to fix this issue.